openwebwork/webwork2
# Hardcopy Themes

This pull creates the ability to have Hardcopy "Themes" and adds a new "oneColumn" theme which prints out homework with one (i.e no) column.   This pull should be tested alongside PG pull:  https://github.com/openwebwork/pg/pull/270

The main things to watch out for: 
-  There were duplicates of the "snippets" directory, so I consolidated everything into webwork2/snippets.  
-  I left links to the old snippet files available to support backwards compatibility.  I also left the option to override the snippet files using conf variables.  (These variables will override any theme choices.) 
-  I added some TeX to PG so that it will check to see if a "nocolumns" flag/method is set before it does anything with multicols.  I may not have done that in the cleanest possible way.  

To test: 
-  Test that you can print out hardcopies in one and two column mode.  Test all three homework types.  
  -  In particular for Gateways I was getting text like "Problem 2. 2." in hardcopies.  I commented out the code which caused it, but I'm not sure why it was there.  My suspicion is that it was added by someone who didn't use ` TEXT(beginproblem());` in their pg files.  
-  Test that you can switch and set the mode using the course configuration menu. 
-  Check that the various instructor features still work (multiple users/multiple sets) and that features are restricted for students.
